### ISSUE TITLE
no-unused-vars: assigning a variable to the return value of a function which takes the variable as an argument

### ISSUE THREAD

**Issue by StickyCube** [eslint/eslint#9148]
**Tell us about your environment**

* **ESLint Version:**
4.5.0

* **Node Version:**
6.11.1

* **npm Version:**
4.6.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
   "rules": {
      "no-unused-vars": 1
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**

no-unused-vars-test.js
```js
var foo = 123;

foo = increment(foo);

function increment (value) {
  return value + 1;
}
```

terminal:
```bash
$ eslint no-unused-vars-test.js
```

**What did you expect to happen?**
variable `foo` is used on line 3: `foo = increment(foo);` so i expect no warnings

**What actually happened? Please include the actual, raw output from ESLint.**
A warning is generated
```
1:5  warning  'foo' is assigned a value but never used  no-unused-vars
```

Note that if the line is changed so that foo is not assigned to, eslint works as expected.

**Comment by not-an-aardvark** [comment]
I think this is working as intended because `foo` is not used again after being assigned to. Your code has an identical effect to:

```js
var foo = 123;

increment(foo);

function increment (value) {
  return value + 1;
}
```

**Comment by StickyCube** [comment]
Yeah i see what you mean. in my actual code i have a variable that is reassigned in this manner 3 or 4 times in the same function. Removing the last assignment removes the warning.

I'll close the issue. Thanks
